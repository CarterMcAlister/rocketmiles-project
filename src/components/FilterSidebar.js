import React from 'react';
import styled from 'styled-components';

import { colors } from '../styleVariables';

const FilterSidebar = ({ sortOrder, setSortOrder, nameFilter, setNameFilter }) => {
    const priceChange = event => {
        setSortOrder(event.target.value);
    };

    const nameFilterChange = event => {
        setNameFilter(event.target.value);
    };

    const resetFilters = () => {
        setNameFilter('');
        setSortOrder('recommended');
    };

    return (
        <FilterSidebarStyled>
            <div className="filters">
                Hotel name
                <input type="text" className="input" value={nameFilter} onChange={nameFilterChange} />
                Price
                <select name="" className="select" onChange={priceChange} value={sortOrder}>
                    <option value="recommended">Recommended</option>
                    <option value="low-to-high">Price low-to-high</option>
                    <option value="high-to-low">Price high-to-low</option>
                </select>
                <button className="button" onClick={resetFilters}>
                    Reset
                </button>
            </div>
        </FilterSidebarStyled>
    );
};

const FilterSidebarStyled = styled.div`
    .filters {
        position: sticky;
        top: 0;
        height: fit-content;
        font-size: 0.8rem;
        color: ${colors.gray};
        background-color: ${colors.white};
        padding: 10px;
        border-radius: 2px;
        box-shadow: 0 2px 7px -2px rgba(0, 0, 0, 0.12);
    }

    .input {
        padding: 0.5rem 1rem;
        font-size: 1rem;
        width: 100%;
        border-radius: 2px;
        border: 1px solid ${colors.lightGray};
        margin-bottom: 10px;
        font-family: system;
        font-weight: 500;
        outline: 0;
        background: white;
        display: inline-block;
    }

    .button {
        width: 100%;
        font-size: 0.8rem;
        border: none;
        outline: none;
        background: ${colors.gray};
        color: white;
        cursor: pointer;
        padding: 10px;
        font-weight: bold;
        border-radius: 2px;
    }

    .select {
        padding: 0.5rem 1rem;
        font-size: 1rem;
        width: 100%;
        border-radius: 2px;
        border: 1px solid ${colors.lightGray};
        margin-bottom: 10px;
        font-family: system;
        font-weight: 500;
        outline: 0;
        background: white;
        display: block;
    }
`;

export default FilterSidebar;
