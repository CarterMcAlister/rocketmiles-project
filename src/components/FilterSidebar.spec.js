import React from 'react';
import { shallow } from 'enzyme';

import FilterSidebar from './FilterSidebar';

describe('FilterSidebar', () => {
    let sortOrderMock = 'recommended';
    const setSortOrderMock = jest.fn(value => (sortOrderMock = value));

    let nameFilterMock = '';
    const setNameFilterMock = jest.fn(value => (nameFilterMock = value));

    const wrapper = shallow(
        <FilterSidebar
            sortOrder={sortOrderMock}
            setSortOrder={setSortOrderMock}
            nameFilter={nameFilterMock}
            setNameFilter={setNameFilterMock}
        />
    );

    it('should display with default values', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should update the sort order when dropdown is changed', () => {
        const newDropdownVal = 'low-to-high';
        wrapper.find('select').simulate('change', { target: { value: newDropdownVal } });

        expect(setSortOrderMock.mock.calls.length).toBe(1);
        expect(sortOrderMock).toEqual(newDropdownVal);
    });

    it('should update the name filter when text is entered', () => {
        const inputText = 'chicago';
        wrapper.find('input').simulate('change', { target: { value: inputText } });

        expect(setNameFilterMock.mock.calls.length).toBe(1);
        expect(nameFilterMock).toEqual(inputText);
    });

    it('should reset the filter values when reset button is clicked', () => {
        wrapper.find('button').simulate('click');

        expect(nameFilterMock).toEqual('');
        expect(sortOrderMock).toEqual('recommended');
    });
});
