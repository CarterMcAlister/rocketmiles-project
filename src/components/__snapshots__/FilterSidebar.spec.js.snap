// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterSidebar should display with default values 1`] = `"<div class=\\"sc-bdVaJa jowjtx\\"><div class=\\"filters\\">Hotel name<input type=\\"text\\" class=\\"input\\" value=\\"\\"/>Price<select name=\\"\\" class=\\"select\\"><option selected=\\"\\" value=\\"recommended\\">Recommended</option><option value=\\"low-to-high\\">Price low-to-high</option><option value=\\"high-to-low\\">Price high-to-low</option></select><button class=\\"button\\">Reset</button></div></div>"`;
